### PR TITLE
clever-tools: init at 3.7.0

### PR DESCRIPTION
**Pull request by florian-sanders-cc**
## Description of changes

Adds the [clever-tools](https://github.com/CleverCloud/clever-tools) package.
This CLI allows users to deploy on Clever Cloud and control their applications, add-ons, services from command line.

Several of Clever Cloud employees actually use NixOS daily and we have a growing number of customers using Nix as well so upstreaming this package would be great.

More employees will be added as maintainers later on so I created a team.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (dani0854) [inline pkgs/by-name/cl/clever-tools/package.nix:1]: `$out/bin` also contains 2 more executables, `install-clever-completion` and `uninstall-clever-completion`.
They appear to be bash scripts which were accidentally wrapped with nodejs, and don't work.

**Review by dani0854 — CHANGES_REQUESTED** [review]
##### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [ ] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [ ] `meta.mainProgram` is set, if applicable.
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] github username matches `githubId`

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix:1]: They are used to install the completions during the `postInstall` hook but I forgot about them.
I'll remove them at the end of the `postInstall` hook, let me know if you think there's a better solution, thanks!

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix:31]: I hesitated between `platforms.all` & linux + darwin. Its a fairly simple Node.js package and I think it should work everywhere Node.js works but I cannot test all platforms myself so I went with what I know is working (linux + darwin), let me know if that's not the right move.

**Comment by florian-sanders-cc** [comment]
Thanks a lot for your time and your feedback.

Everything should be fixed but don't hesitate to tell me if we can improve stuff.

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (dani0854) [reply pkgs/by-name/cl/clever-tools/package.nix:33]: Seems like buildNpmPackage sets platforms by default to the interpreter supported platforms, here:
https://github.com/NixOS/nixpkgs/blob/3a729c132a7a120b53d51ec3d4aacfb83aa665cd/pkgs/build-support/node/build-npm-package/default.nix#L87

Which in this case would still be linux + darwin. So I guess the best way would be actually to not specify `meta.platforms` after all, unless it is a subset of what the interpreter supports.

Sorry, my bad)

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (dani0854) [reply pkgs/by-name/cl/clever-tools/package.nix:1]: I don't know what would be the best practice here.

According to the documentation, during install, it uses `npm pack --json --dry-run` to decide which files to wrap. Which relies on `package.json` `bin` section.

Also, since you use `clever` executable to generate completions in `postInstall`, those scripts probably aren't needed at all.

I can think of 2 ways to remove them, either patch `pakage.json` before install, or remove wrappers in `postInstall`.

The latter seems to be a much simpler solution, so I personally think it's the best solution for a simple package.

But other reviewers may have a different opinion.

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix:33]: No worries, I'll change that.

Thanks for the explanation, I should have checked it before submitting the PR :+1:

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix:1]: Alright I see, I'll discuss it with the CLI team, we may be able to remove it from the source since it's probably not used by anyone and very legacy.

Anyway, even if we do remove these from source, it won't be included within this version so it would be great if we could merge it as it is now.

**Review by dani0854 — APPROVED** [review]

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (jtbx) [inline pkgs/by-name/cl/clever-tools/package.nix:17]: Contributing guide advises against this
```suggestion
    rev = version;
```

**Review by jtbx — CHANGES_REQUESTED** [review]

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix]: Bad habits die hard, this should be fixed now, thanks!

**Review by jtbx — APPROVED** [review]
Thanks for that, looks good :+1:

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (Aleksanaa) [inline pkgs/by-name/cl/clever-tools/package.nix:27]: We'd better use `installShellCompletion`, `mkdir` is also not needed.

Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix]: Fixed, thanks I did not know about this one and it is indeed a lot better this way :+1: 

Seems like I've picked up formatting issues when rebasing, don't really know why but I'll check tomorrow to fix CI checks.

**Comment by florian-sanders-cc** [comment]
Everything should be fixed, I'm not quite sure what to do about `team-list.nix` formatting.
I've formatted the file to make it pass CI Checks, don't hesitate to tell me if it needs to be reverted :+1:

File: maintainers/team-list.nix
Comment (dani0854) [inline maintainers/team-list.nix:1]: It seems like there is a lot of formatting changes in `team-list.nix`.

The commit bellow has formatted that file. Do a rebase against `master` branch.
https://github.com/NixOS/nixpkgs/commit/5cf83a58da3ca8b51f70ce4c5ed8bbad4304586f

**Comment by florian-sanders-cc** [comment]
@NixOS/nixpkgs-merge-bot merge

**Comment by florian-sanders-cc** [comment]
Figured it wouldn't work before this PR was merged but was worth the try cause I didn't want to bother anyone. Sorry for the spam :wink:

**Comment by dani0854** [comment]
Looks like this PR is waiting for @Aleksanaa review, or I don't quite understand why `approvals: 2` label got removed.

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (rhendric) [inline pkgs/by-name/cl/clever-tools/package.nix:39]:
```suggestion
    description = "Deploy on Clever Cloud and control your applications, add-ons, services from command line";
```

File: pkgs/by-name/cl/clever-tools/package.nix
Comment (rhendric) [inline pkgs/by-name/cl/clever-tools/package.nix:42]: Did you want to use `teams.clevercloud.members` here?

Comment (florian-sanders-cc) [reply pkgs/by-name/cl/clever-tools/package.nix]: Yes definitely, thanks!

**Review by rhendric — APPROVED** [review]

**Pull request merged**
